kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/ResourceDetailsBottomSheet.kt

```diff
@@ -129,6 +89,90 @@ class ResourceDetailsBottomSheet(private val resource: ViewResource) : BottomShe
         }
     }
 
+    private fun resourceToggleText(): String {
+        if (resource.enabled) {
+            return "Disable this resource"
+        } else {
+            return "Enable this resource"
+        }
+    }
+
+    private fun resourceHeader() {
+        if (resource.isInternetResource()) {
+            internetResourceHeader()
+        } else {
+            nonInternetResourceHeader()
+        }
+    }
+
+    private fun internetResourceHeader() {
+        val addToFavoritesBtn: MaterialButton = view.findViewById(R.id.addToFavoritesBtn)
+        val removeFromFavoritesBtn: MaterialButton = view.findViewById(R.id.removeFromFavoritesBtn)
+        val resourceNameTextView: TextView = view.findViewById(R.id.tvResourceName)
+        val resourceAddress: LinearLayout = view.findViewById(R.id.addressSection)
+        val resourceAddressDescriptionTextView: TextView = view.findViewById(R.id.tvResourceAddressDescription)
+        val resourceDescriptionLayout: LinearLayout = view.findViewById(R.id.resourceDescriptionLayout)
+
+        addToFavoritesBtn.visibility = View.GONE
+        removeFromFavoritesBtn.visibility = View.GONE
+
+        resourceNameTextView.text = resource.name
+
+        resourceAddress.visibility = View.GONE
+
+        resourceDescriptionLayout.visibility = View.VISIBLE
+        resourceAddressDescriptionTextView.text = "All network traffic"
+
+    }
+
+    private fun nonInternetResourceHeader() {
+        val addToFavoritesBtn: MaterialButton = view.findViewById(R.id.addToFavoritesBtn)
+        val removeFromFavoritesBtn: MaterialButton = view.findViewById(R.id.removeFromFavoritesBtn)
+        val resourceNameTextView: TextView = view.findViewById(R.id.tvResourceName)
+        val resourceAddressTextView: TextView = view.findViewById(R.id.tvResourceAddress)
+        val resourceAddressDescriptionTextView: TextView = view.findViewById(R.id.tvResourceAddressDescription)
+        val resourceDescriptionLayout: LinearLayout = view.findViewById(R.id.resourceDescriptionLayout)
+
+
+        addToFavoritesBtn.setOnClickListener {
+            viewModel.addFavoriteResource(resource.id)
+            refreshButtons()
+        }
+        removeFromFavoritesBtn.setOnClickListener {
+            viewModel.removeFavoriteResource(resource.id)
+            refreshButtons()
+        }
+        refreshButtons()
+
+        resourceNameTextView.text = resource.name
+        val displayAddress = resource.addressDescription ?: resource.address
+        resourceAddressTextView.text = displayAddress
+
+        if (!resource.addressDescription.isNullOrEmpty()) {
+            resourceAddressDescriptionTextView.text = resource.addressDescription
+            resourceDescriptionLayout.visibility = View.VISIBLE
```

Comment:
Until this PR the resource description wasn't showing on Android, since the visibility was being changed in the text not the whole layout which is the hidden one.

@jamilbk was this on purpose?

Reply:
I don't think so, that was probably a bug. The intended behavior is that if the addressDescription is available show that, otherwise show address.

Reply:
Resources don't have a Description field. Only Address description, which is used to display things like `https://github.com` or `1.1.1.1:22`. https://www.firezone.dev/kb/deploy/resources#address-description

Reply:
Actually, yes, that's intended. We don't show a dedicated field for the description, right? It's `addressDescription || address`. The address_description replaces the address shown if the admin has entered one.

Reply:
We should probably then remove the element and this whole block

Reply:
I mean right now in main this element is modified but never made visible at all.

We can otherwise hide the address field when this one is active

Reply:
Hm yeah if you wouldn't mind the intended behavior is working that would be much appreciated! I had tested that but it may have been changed in the last few rounds of changes.

Reply:
Yeah, it was working, this few lines were just doing nothing, so I removed them, kept the hidden layout for the internet description so that we don't have to modify the label